dm/master/server.go

```diff
@@ -1781,27 +1768,140 @@ func (s *Server) OperateSchema(ctx context.Context, req *pb.OperateSchemaRequest
 	}, nil
 }
 
-// GetTaskCfg implements MasterServer.GetSubTaskCfg
-func (s *Server) GetTaskCfg(ctx context.Context, req *pb.GetTaskCfgRequest) (*pb.GetTaskCfgResponse, error) {
+func (s *Server) createMasterClientByName(ctx context.Context, name string) (pb.MasterClient, error) {
+	listResp, err := s.etcdClient.MemberList(ctx)
+	if err != nil {
+		return nil, err
+	}
+	clientURLs := []string{}
+	for _, m := range listResp.Members {
+		if name == m.Name {
+			for _, url := range m.GetClientURLs() {
+				clientURLs = append(clientURLs, utils.UnwrapScheme(url))
+			}
+			break
+		}
+	}
+	if len(clientURLs) == 0 {
+		return nil, errors.New("master not found")
+	}
+	tls, err := toolutils.NewTLS(s.cfg.SSLCA, s.cfg.SSLCert, s.cfg.SSLKey, s.cfg.AdvertiseAddr, s.cfg.CertAllowedCN)
+	if err != nil {
+		return nil, err
+	}
+
+	var conn *grpc.ClientConn
+	for _, clientURL := range clientURLs {
+		//nolint:staticcheck
+		conn, err = grpc.Dial(clientURL, tls.ToGRPCDialOption(), grpc.WithBackoffMaxDelay(3*time.Second))
+		if err == nil {
+			masterClient := pb.NewMasterClient(conn)
+			return masterClient, nil
+		}
+		log.L().Error("can not dial to master", zap.String("name", name), zap.String("client url", clientURL), log.ShortError(err))
+	}
+	// return last err
+	return nil, err
+}
+
+// GetMasterCfg implements MasterServer.GetMasterCfg
+func (s *Server) GetMasterCfg(ctx context.Context, req *pb.GetMasterCfgRequest) (*pb.GetMasterCfgResponse, error) {
+	log.L().Info("", zap.Any("payload", req), zap.String("request", "GetMasterCfg"))
+
+	var err error
+	resp := &pb.GetMasterCfgResponse{}
+	resp.Cfg, err = s.cfg.Toml()
+	return resp, err
+}
+
+// GetCfg implements MasterServer.GetCfg
+func (s *Server) GetCfg(ctx context.Context, req *pb.GetCfgRequest) (*pb.GetCfgResponse, error) {
 	var (
-		resp2 *pb.GetTaskCfgResponse
+		resp2 = &pb.GetCfgResponse{}
 		err2  error
+		cfg   string
 	)
 	shouldRet := s.sharedLogic(ctx, req, &resp2, &err2)
 	if shouldRet {
 		return resp2, err2
 	}
 
-	cfg := s.scheduler.GetTaskCfg(req.Name)
+	switch req.Type {
+	case pb.CfgType_TaskType:
+		subCfgMap := s.scheduler.GetSubTaskCfgsByTask(req.Name)
+		if len(subCfgMap) == 0 {
+			resp2.Msg = "task not found"
+			return resp2, nil
+		}
+		subCfgList := make([]*config.SubTaskConfig, 0, len(subCfgMap))
+		for _, subCfg := range subCfgMap {
+			subCfgList = append(subCfgList, subCfg)
+		}
+		sort.Slice(subCfgList, func(i, j int) bool {
+			return subCfgList[i].SourceID < subCfgList[j].SourceID
+		})
 
-	if len(cfg) == 0 {
-		return &pb.GetTaskCfgResponse{
-			Result: false,
-			Msg:    "task not found",
-		}, nil
+		var taskCfg config.TaskConfig
+		taskCfg.FromSubTaskConfigs(subCfgList...)
```

Comment:
How about move reciever of this function to return value
And update the comment of this function (not only used when import from v1.0)